### PR TITLE
VT-3665 MPC Enhancements for Go SDK

### PR DESCRIPTION
**Pull request by Abinaya-Shunmugavel**

### CODE REVIEW

File: multipartycall_test.go [hunk 1]
```diff
@@ -135,3 +135,61 @@ func TestMPCService_GetParticipant(t *testing.T) {
 
 	assertRequest(t, "GET", "MultiPartyCall/%s/Participant/%s", "uuid_ebacced2-21ab-466d-9df4-67339991761b", "209")
 }
+
+func TestMPCService_StartParticipantRecord(t *testing.T) {
+	expectResponse("MPCStartParticipantRecordResponse.json", 202)
```
Comment (Koushik-Ayila) [inline multipartycall_test.go:140]: instead of 202, it should be 200

File: multipartycall_test.go [hunk 2]
```diff
@@ -135,3 +135,61 @@ func TestMPCService_GetParticipant(t *testing.T) {
...
+	if err := client.MultiPartyCall.PauseParticipantRecording(MultiPartyCallParticipantParams{MpcUuid: "ebacced2-21ab-466d-9df4-67339991761b", ParticipantId: "209"}); err != nil {
+		panic(err)
+	}
+
+	assertRequest(t, "POST", "MultiPartyCall/%s/Participant/%s/Record/Pause", "uuid_ebacced2-21ab-466d-9df4-67339991761b", "209")
+}
+func TestMPCService_StartPlayAudio(t *testing.T) {
+	expectResponse("MPCStartPlayAudioResponse.json", 202)
+
+	if response, err := client.MultiPartyCall.StartPlayAudio(MultiPartyCallParticipantParams{MpcUuid: "ebacced2-21ab-466d-9df4-67339991761b", ParticipantId: "209"}, MultiPartCallAudio{Url: "https://s3.amazonaws.com/plivocloud/music.mp3"}); err != nil {
```
Comment (Koushik-Ayila) [inline multipartycall_test.go:179]: Please make UT change here as well similar to that of code

File: xml/plivoxml_test.go [hunk 3]
```diff
@@ -61,9 +61,9 @@ func TestComplexXML(t *testing.T) {
...
+			new(MultiPartyCallElement).SetAgentHoldMusicMethod("GET").SetCoachMode(true).SetCustomerHoldMusicMethod("GET").SetEndMpcOnExit(false).SetEnterSound("beep:1").SetEnterSoundMethod("GET").SetExitSound("beep:2").SetExitSoundMethod("GET").SetHold(false).SetMaxDuration(14400).SetMaxParticipants(10).SetMute(false).SetOnExitActionMethod("POST").SetRecord(false).SetRecordFileFormat("mp3").SetRecordingCallbackMethod("POST").SetRelayDtmfInputs(false).SetRole("agent").SetStartMpcOnEnter(true).SetStatusCallbackEvents("mpc-state-changes,participant-state-changes").SetStatusCallbackMethod("POST").SetStayAlone(false).SetWaitMusicMethod("GET").SetContents("Nairobi").SetStartRecordingAudio("beep:3").SetStartRecordingAudioMethod("GET").SetStopRecordingAudio("beep:4").SetStopRecordingAudioMethod("GET"),
```
Comment (Koushik-Ayila) [inline xml/plivoxml_test.go:66]: StartRecordingAudio and StopRecordingAudio should be URLs

**Review by Koushik-Ayila — CHANGES_REQUESTED** [review]
Also, please add the renamed mock response files as well

File: multipartycall.go [hunk 4]
```diff
@@ -184,6 +188,17 @@ type MultiPartyCallUpdateParticipantResponse struct {
 	MPCUpdateResponse
 }
 
+type MultiPartCallAudioParams struct {
```
Comment (Koushik-Ayila) [inline multipartycall.go:191]: rename this to MultiPartyCallAudioParams

File: multipartycall_test.go [hunk 5]
```diff
@@ -135,3 +135,61 @@ func TestMPCService_GetParticipant(t *testing.T) {
...
+	if err := client.MultiPartyCall.PauseParticipantRecording(MultiPartyCallParticipantParams{MpcUuid: "ebacced2-21ab-466d-9df4-67339991761b", ParticipantId: "209"}); err != nil {
+		panic(err)
+	}
+
+	assertRequest(t, "POST", "MultiPartyCall/%s/Participant/%s/Record/Pause", "uuid_ebacced2-21ab-466d-9df4-67339991761b", "209")
+}
+func TestMPCService_StartPlayAudio(t *testing.T) {
+	expectResponse("MPCStartPlayAudioResponse.json", 202)
+
+	if response, err := client.MultiPartyCall.StartPlayAudio(MultiPartyCallParticipantParams{MpcUuid: "ebacced2-21ab-466d-9df4-67339991761b", ParticipantId: "209"}, MultiPartCallAudioParams{Url: "https://s3.amazonaws.com/plivocloud/music.mp3"}); err != nil {
```
Comment (Koushik-Ayila) [inline multipartycall_test.go:179]: please rename here as well

File: xml/plivoxml_test.go [hunk 6]
```diff
@@ -61,9 +61,9 @@ func TestComplexXML(t *testing.T) {
...
+			new(MultiPartyCallElement).SetAgentHoldMusicMethod("GET").SetCoachMode(true).SetCustomerHoldMusicMethod("GET").SetEndMpcOnExit(false).SetEnterSound("beep:1").SetEnterSoundMethod("GET").SetExitSound("beep:2").SetExitSoundMethod("GET").SetHold(false).SetMaxDuration(14400).SetMaxParticipants(10).SetMute(false).SetOnExitActionMethod("POST").SetRecord(false).SetRecordFileFormat("mp3").SetRecordingCallbackMethod("POST").SetRelayDtmfInputs(false).SetRole("agent").SetStartMpcOnEnter(true).SetStatusCallbackEvents("mpc-state-changes,participant-state-changes").SetStatusCallbackMethod("POST").SetStayAlone(false).SetWaitMusicMethod("GET").SetContents("Nairobi").SetStartRecordingAudio("url").SetStartRecordingAudioMethod("GET").SetStopRecordingAudio("url").SetStopRecordingAudioMethod("GET"),
```
Comment (Koushik-Ayila) [inline xml/plivoxml_test.go:66]: url would mean some valid url

**Review by Koushik-Ayila — CHANGES_REQUESTED** [review]

**Review by Koushik-Ayila — APPROVED** [review]

**Review by varshit97plivo — APPROVED** [review]

**Review by varshit97plivo — APPROVED** [review]

**Pull request merged**